hbase-common/src/test/java/org/apache/hadoop/hbase/io/TestByteBuffAllocator.java

```diff
@@ -21,6 +21,7 @@
 import static org.apache.hadoop.hbase.io.ByteBuffAllocator.getHeapAllocationRatio;
```

Comment:
Would you consider a new unit here that confirms leak detection works, by leaking deliberately?

Reply:
Not a bad idea. I took a look and unfortunately netty's ResourceLeakDetector doesn't really give us a way to hook into the detection in a way that we could test. Maybe there's a way to intercept log4j logs? If you have an easy idea for that, would be happy to try it.

Reply:
You can call InternalLoggerFactory.setDefaultFactory to inject a customized InternalLogger which will be used in netty.